src/number.cr

```diff
@@ -47,7 +47,7 @@ struct Number
   macro slice(*nums, read_only = false)
     %slice = Slice({{@type}}).new({{nums.size}}, read_only: {{read_only}})
     {% for num, i in nums %}
-      %slice.to_unsafe[{{i}}] = {{@type}}.new({{num}})
+      %slice.to_unsafe[{{i}}] = {{@type}}.new!({{num}})
```

Comment:
I'm thinking this should raise on overflow. Same with `static_array`. That it works now and there's a test for it is just because it was working like that so far.

Reply:
There is a spec that explicitly use out of range values. We either break introduce a breaking change (that maybe nobody cares) or we keep the current behavior.

I agree that it's odd to allow negative numbers in a `UInt8.slice(-1)`. I just didn't want to introduce breaking changes here.

Reply:
If it's to avoid a breaking change then yes, let this silently overflow for now 👍